ocaml/testsuite/tests/typing-unique/overwriting_lift_constants_expect.ml

```diff
@@ -0,0 +1,206 @@
+(* TEST
+   flags += "-extension unique ";
+   flags += "-extension overwriting ";
+   expect;
+*)
+
+type point = { dim : int; x : float; y : float; z : float }
+[%%expect{|
+type point = { dim : int; x : float; y : float; z : float; }
+|}]
+
+(* This file tests how constants are lifted out.
+   Unique constants can not be lifted out since they
+   might be overwritten but all other constants should still
+   be lifted out.
+   Since this pass is not yet implemented, I have added the
+   expected result from the prototype in a comment to each test.
+*)
+
+(* When overwriting p with constants, this constant can be lifted out *)
+let update_with_constant (p : point) =
+  let q = overwrite_ p with { dim = 3; x = 2.0; y = 3.0; z = 4.0 } in
+  q
+[%%expect{|
+Line 2, characters 10-66:
+2 |   let q = overwrite_ p with { dim = 3; x = 2.0; y = 3.0; z = 4.0 } in
+              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alert : Overwrite not implemented.
+Uncaught exception: File "ocaml/parsing/location.ml", line 1106, characters 2-8: Assertion failed
+
+|}]
+(*
+[%%expect{|
+val update_with_constant : unique_ point -> point = <fun>
+|}]
+*)
+
+(* In this test, the update function returns the same value each time *)
+let test =
+  let p = { dim = 3; x = 1.0; y = 2.0; z = 3.0 } in
+  let q = { dim = 3; x = 1.0; y = 2.0; z = 3.0 } in
+  update_with_constant p == update_with_constant q
+[%%expect{|
+Line 4, characters 2-22:
+4 |   update_with_constant p == update_with_constant q
+      ^^^^^^^^^^^^^^^^^^^^
+Error: Unbound value "update_with_constant"
+|}]
+(*
+[%%expect{|
+val test : bool = true
```

Comment:
If I'm not mistaken, `p` and `q` are shared because they are immutable and equal, but since they are unique that should be forbidden. Ultimately I think this issue should be resolved when `alloc_mode` properly tracks uniqueness, in particular mapping immutable but unique allocations to `Immutable_unique` instead of `Immutable` in the middle-end.

Reply:
Indeed, my expectation was different: I vaguely remembered that my patch used to lift out `Preuseblocks` that are fully overwritten with constants. That would then cause `update_with_constant` to always return the same record no matter the input (and the fact that both `p` and `q` are the same would just be a distraction). But looking at `middle_end/flambda/inconstant_idents.ml` in https://github.com/ocaml-flambda/ocaml-jst/pull/33/files, I see that I created a CR-someday for that functionality so now I am confused as to why I used to get that behaviour.

Reply:
Is there still an open question here?

Reply:
I think this test might have always been broken, since I also had it in an expect-test before that should not be able to even perform reuse (and perhaps even lifting?). So I think we should just go with the new test and meet again if our backend implementation yields a different output. I have added plenty comments to the new file so that if the output is different, others can check my reasoning for why I believe we should get that output.